### PR TITLE
Tooltip outside click, should abort share creation

### LINKED ISSUE #42046: [Bug]: Year cannot be typed correctly in sharing flow
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

When you enter a date in the sharing expire date field through the web interface, the calculated year is unreasonable (starting from 1900).

https://github.com/nextcloud/server/assets/70155116/c0d97ea7-ef00-4537-b373-cb4803d75da4

### Steps to reproduce

1. Create new share link
2. Enable expire date
3. Select year in expire date and start typing numbers

### Expected behavior

The year should match the user input (i.e. if I have typed "2" it should show "2", instead of "1902"

### Installation method

None

### Nextcloud Server version

28

### Operating system

Debian/Ubuntu

### PHP engine version

PHP 8.3

### Web server

Apache (supported)

### Database engine version

MariaDB

### Is this bug present after an update or on a fresh install?

Fresh Nextcloud Server install

### Are you using the Nextcloud Server Encryption module?

Encryption is Disabled

### What user-backends are you using?

- [X] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

_No response_

### List of activated Apps

_No response_

### Nextcloud Signing status

_No response_

### Nextcloud Logs

_No response_

### Additional info

_No response_

### PR DESCRIPTION
**Pull request by nfebe**
- It's unexpected that click outside a tooltip would proceed with the action that could be carried out inside the tooltip.
- Allow user to finish typing date before formatting

Resolves : https://github.com/nextcloud/server/issues/42046, https://github.com/nextcloud/server/issues/42114

### CODE REVIEW

**Review by artonge — CHANGES_REQUESTED** [review]
Some files seem unrelated

File: apps/files_trashbin/tests/js/appSpec.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
+/**
+* @copyright 2014 Vincent Petry <pvince81@owncloud.com>
+ *
+ * @author Vincent Petry <vincent@nextcloud.com>
```
[the remark this answers is not in the document]
Comment (Fenn-CS) [reply apps/files_trashbin/tests/js/appSpec.js]: Thank you!

**Review by artonge — APPROVED** [review]

**Review by susnux — APPROVED** [review]

**Comment by Fenn-CS** [comment]
backport to stable28

**Comment by Fenn-CS** [comment]
backport to stable27

**Comment by skjnldsv** [comment]
/skjnldsv-backport c179408a0dd8710f7b515179aab3dd9407794b0c 04e87337211351127cd02a0404936de8cb1b0837 to stable28

**Pull request merged**